crates/sozo/src/ops/component.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::Result;
+use dojo_world::world::WorldContractReader;
+use starknet::core::types::{BlockId, BlockTag};
+use toml::Value;
+
+use crate::commands::component::ComponentCommands;
+
+pub async fn execute(command: ComponentCommands, env_metadata: Option<Value>) -> Result<()> {
+    match command {
+        ComponentCommands::Get { name, world, starknet } => {
+            let world_address = world.address(env_metadata.as_ref())?;
+            let provider = starknet.provider(env_metadata.as_ref())?;
+
+            let world = WorldContractReader::new(world_address, &provider);
+            let component = world.component(&name, BlockId::Tag(BlockTag::Pending)).await?;
+
+            println!("{:#x}", component.class_hash());
+        }
+
+        ComponentCommands::Schema { name, world, starknet, to_json } => {
+            let world_address = world.address(env_metadata.as_ref())?;
+            let provider = starknet.provider(env_metadata.as_ref())?;
+
+            let world = WorldContractReader::new(world_address, &provider);
+            let component = world.component(&name, BlockId::Tag(BlockTag::Pending)).await?;
+
+            let schema = component.schema(BlockId::Tag(BlockTag::Pending)).await?;
+
+            if to_json {
+                println!("{}", serde_json::to_string_pretty(&schema)?)
+            } else {
+                let output = format!(
+                    r"struct {name} {{
+{}
+}}",
+                    schema
+                        .iter()
+                        .map(|s| format!(r"   {}: {}", s.name, s.ty))
+                        .collect::<Vec<String>>()
+                        .join("\n")
+                );
+
+                println!("{output}")
+            }
+        }
+
+        ComponentCommands::Entity { .. } => {
```

Comment:
@tarrencev how should the params for the `entity` command look like?

Reply:
it should take:
a key array (Vec<FieldElement>)
a partition id

so something like

```
sozo component key0
sozo component "[key0, key1]"
sozo component key0 --parition foo
```